src/main/java/com/webdev/cosmo/cosmobackend/posts/controller/PostController.java

```diff
@@ -0,0 +1,61 @@
+package com.webdev.cosmo.cosmobackend.posts.controller;
+
+import com.webdev.cosmo.cosmobackend.posts.dto.PostDTO;
+import com.webdev.cosmo.cosmobackend.posts.exception.NotFoundException;
+import com.webdev.cosmo.cosmobackend.posts.service.PostService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/posts")
+@RequiredArgsConstructor
+public class PostController {
+    private final PostService service;
+
+    @PostMapping
+    public ResponseEntity<PostDTO> createPost(@RequestBody PostDTO post) {
+        PostDTO savedPost = service.createPost(post);
+        return new ResponseEntity<>(savedPost, HttpStatus.CREATED);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<PostDTO> getPostById(@PathVariable("id") Long postId) {
```

Comment:
You dont have to specify "id" in pathVariable, just use @GetMapping("/{postId}")